app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -62,6 +50,40 @@ protected void onCreate(Bundle savedInstanceState) {
 
         locationClient = new LocationClient(this);
 
+        weatherServiceCallback = new IWeatherServiceCallback.Stub() {
```

Comment:
no need to have those inner classes declaration/field initialization in onCreate(). move out to the field initialization.

Reply:
1. weatherServiceCallback initialization I moved out to declaration section.
2. locationClient constructor has input parameter reference to MainActivity, and in time of locationClient's declaration MainActivity still not constructed. Therefore locationClient's  initializatin are in onCreate method.